src/server/ExpressHttpServer.ts

```diff
@@ -1,26 +1,47 @@
 import type { Server } from 'http';
+import type { Socket } from 'net';
 import cors from 'cors';
 import type { Express } from 'express';
 import express from 'express';
+import type WebSocket from 'ws';
+import { Server as WebSocketServer } from 'ws';
 import { getLoggerFor } from '../logging/LogUtil';
 import type { HttpHandler } from './HttpHandler';
+import type { HttpRequest } from './HttpRequest';
+import type { WebSocketHandler } from './WebSocketHandler';
 
 export class ExpressHttpServer {
   protected readonly logger = getLoggerFor(this);
 
-  private readonly handler: HttpHandler;
+  private readonly httpHandler: HttpHandler;
+  private readonly webSocketHandler: WebSocketHandler;
 
-  public constructor(handler: HttpHandler) {
-    this.handler = handler;
+  public constructor(httpHandler: HttpHandler, webSocketHandler: WebSocketHandler) {
+    this.httpHandler = httpHandler;
+    this.webSocketHandler = webSocketHandler;
   }
 
   public listen(port?: number): Server {
```

Comment:
The way it is written now, we only provide 1 HttpServer: one that does both the HTTP part and the websocket part, there is no way to set up a server without websockets anymore. I was thinking of perhaps keeping the `ExpressHttpServer` as it was, and creating a new `WebSocketHttpServer` where we inject the `ExpressHttpServer`? The `listen` function of the interface already returns a server, so that server could call `listen` on the server it gets as input and then attach the websocket functionality to that.

Reply:
Understood; and would fit nicely with our modularity perspective. The nice thing is that then for the whole WebSocket thing, I would literally not have to change a single line of code but only add 🙂

However, I thought it might be overkill for this situation, creating more moving pieces that increase complexity rather than removing it. I also created the `RejectWebSocketHandler` so we could disable.

Reply:
I'm just thinking for the future if someone ever wants to have a koa server because they really don't want express or other reasons, they would have to reimplement the ws part. Doesn't seem like that much overkill either since that part of the config is probably going to stay as-is mostly. So I'm in favour of doing it like that.

Reply:
Good point, will refactor like this.